packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -83,142 +88,65 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
-        // Double braces like `{{` are Rust's escape string for a single literal `{`.
-        let wrapped_code = new_js_string(
-            scope,
-            &format!(
-                "((hash_util, hash_stdlib) => {{{code}
-                    return [
-                        typeof start_experiment === 'undefined' ? undefined : start_experiment,
-                        typeof start_sim === 'undefined' ? undefined : start_sim,
-                        typeof run_task === 'undefined' ? undefined : run_task
-                    ]
-                }})",
-            ),
-        );
-
-        let pkg: Function<'_> = {
-            let mut try_catch = v8::TryCatch::new(scope);
-
-            let pkg = v8::Script::compile(&mut try_catch, wrapped_code, None).ok_or_else(|| {
-                let exception = try_catch.exception().unwrap();
-                let exception_string = exception
-                    .to_string(&mut try_catch)
-                    .unwrap()
-                    .to_rust_string_lossy(&mut try_catch);
-
-                Error::PackageImport(
-                    path.clone(),
-                    format!("Could not compile code for package: {exception_string}"),
-                )
-            })?;
-
-            let pkg = pkg.run(&mut try_catch).ok_or_else(|| {
-                let exception = try_catch.exception().unwrap();
-                let exception_string = exception
-                    .to_string(&mut try_catch)
-                    .unwrap()
-                    .to_rust_string_lossy(&mut try_catch);
-
-                Error::PackageImport(
-                    path.clone(),
-                    format!("Couldn't execute package setup: {exception_string}"),
-                )
-            })?;
-
-            pkg.try_into().map_err(|err| {
-                Error::PackageImport(
-                    path.clone(),
-                    format!("Could not convert package setup return value into Function: {err}"),
-                )
-            })?
-        };
-
-        let args = &[embedded.hash_util, embedded.hash_stdlib];
-        let fns: Array<'_> = {
-            let global_context = context.global(scope);
-            let fns = call_js_function(scope, pkg, global_context.into(), args).map_err(|err| {
-                Error::PackageImport(path.clone(), format!("Couldn't call package: {err}"))
-            })?;
-
-            fns.try_into().map_err(|err| {
-                Error::PackageImport(
-                    path.clone(),
-                    format!(
-                        "Couldn't convert package return value to array of package functions: \
-                         {err}"
-                    ),
-                )
-            })?
-        };
+        let fn_names = ["start_experiment", "start_sim", "run_task"];
+        let fns = fn_names
+            .into_iter()
+            .map(|fn_name| {
+                let mut try_catch_scope = v8::TryCatch::new(scope);
```

Comment:
Is it concerning performance-wise that we call `TryCatch::new` for each function? Is it possible to re-use a try-catch block?

Reply:
It doesn't look expensive, mostly pointer conversions.
We can use a `TryCatch` as long as we want. If we check for exception regularly I don't think there is much difference between multiple `TryCatch` and a single big one.

For that specific case it was nice to have it in here to not have to drop it manually.
And this is only called once for the performance aspect.

Reply:
I'm fine with having it here, just curious about the reasoning.

> For that specific case it was nice to have it in here to not have to drop it manually.

Does dropping have any implications?

Also, I can't imagine this is just about pointer conversion because having an exception requires the ability to unwind the stack, so this most likely introduces a few compiler intrinsics.

Reply:
As it's a one-time call I think it's not worth investigating it further.